src/components/RelativeTimeText.tsx

```diff
@@ -24,24 +25,45 @@ const locale: Locale = {
   formatRelative: (token) => formatRelativeLocale[token],
 };
 
+export const AbsoluteDateFormat: string = 'EEE yyyy-MM-dd HHmm';
+export const TextBoxDateFormat: string = "yyyy-MM-dd'T'HH:mm";
+
+export enum RelativeStyle {
+  Relative,
+  Auto,
+  Absolute,
+}
+
 export interface RelativeTimeTextProps {
   time: number;
   baseTime: number;
   lowercase?: boolean;
-  defaultToTime?: boolean;
+  relative: RelativeStyle;
 }
 
-export const RelativeTimeText = ({ time, baseTime, defaultToTime, lowercase }: RelativeTimeTextProps) => {
-  const [useRelative, setUseRelative] = React.useState<boolean>(!(defaultToTime ?? false));
+export const formatTimeAsString = (time: number, baseTime: number, useRelative: boolean, lowercase?: boolean) => {
   let text;
   if (useRelative) {
     text = formatRelative(time, baseTime, { locale });
     if (!lowercase) {
       text = text[0].toLocaleUpperCase() + text.substring(1);
     }
   } else {
-    text = formatDate(time, 'EEE yyyy-MM-dd HHmm');
+    text = formatDate(time, AbsoluteDateFormat);
   }
+  return text;
+};
+
+export const RelativeTimeText = ({ time, baseTime, relative, lowercase }: RelativeTimeTextProps) => {
```

Comment:
Give baseTime a default so we don't have to pass in `new Date().getTime()` everywhere:
```
... { time, baseTime = new Date().getTime(), relative, lowercase } ...
```